tests/test_datafiles_install.py

```diff
@@ -0,0 +1,87 @@
+
+
+import pytest
+import subprocess
+import shutil
+import pathlib
+import sys
+
+
+
+data_files_combinations = [
+    (
+        # data file source
+        ("share/test.txt",),
+        # data file spec
+        ("jupyter-packaging-test", "share", "**/*"),
+        # data file target
+        "jupyter-packaging-test/test.txt"
+    ),
+    (
+        # data file source
+        ("share/test.txt",),
+        # data file spec
+        ("jupyter-packaging-test/level1", "share", "**/*"),
+        # data file target
+        "jupyter-packaging-test/level1/test.txt"
+    ),
+    (
+        # data file source
+        ("level1/test/test.txt",),
+        # data file spec
+        ("jupyter-packaging-test", "level1/test", "**/*"),
+        # data file target
+        "jupyter-packaging-test/test.txt"
+    ),
+    (
+        # data file source
+        ("level1/test/test.txt",),
+        # data file spec
+        ("jupyter-packaging-test/level1/level2", "level1/test", "**/*"),
+        # data file target
+        "jupyter-packaging-test//level1/level2/test.txt"
+    ),
+    (
+        # data file source
+        ("level1/level2/test/test.txt",),
+        # data file spec
+        ("jupyter-packaging-test", "level1", "**/*"),
+        # data file target
+        "jupyter-packaging-test/level2/test/test.txt"
```

Comment:
Ok, @blink1073, I think I've covered the recursive case here. Let me know what you think.